---
title: "Infant fecal pH/redox analysis"
output: html_notebook
---

# Load libraries
```{r}
# Plotting
library(tidyverse)
#library(RColorBrewer)
library(ggtext)
library(ggsignif)
library(cowplot)

# Stats
library(phyloseq)
library(lme4)
library(lmerTest)
library(vegan)
library(ALDEx2)

theme_set(theme_bw() +
            theme(axis.text = element_text(color = "black"),
                  axis.ticks = element_line(color = "black"),
                  plot.title = element_text(hjust = 0.5)))

set.seed(123)


```

# Load and clean up data 
```{r}
# pH and redox with clinical data
## See data/variable_descriptions_for_fecal_pH_final.xlsx for explanations of variables
pH_redox <- read.csv("data/fecal_pH_final.csv")

# SCFA data by GC
## Isobutyrate, isovalerate, and valerate were not detected for any samples
## Two samples have been omitted where nothing was detected
## All values have been multiplied by 10 (dilution factor) and are in mM
## Where values were NA/0, they have been replaced with a pseudocount equal to the lowest value observed for that SCFA
gc <- read.csv("data/nicu_gc_cleaned.csv")

# Combine
pH_redox <- pH_redox %>%
  left_join(gc) %>%
  relocate(acetate:total, .after = redox)


```

## 16S
```{r}
# Load phyloseq object, omitting samples with < 5000 reads
## This version used to calculate diversity
ps <- readRDS("data/phyloseq.rds") %>%
  prune_samples(sample_sums(.) >= 5000, .)
# 95 taxa by 77 samples (none removed by filter)

# Omit low abundance taxa for differential abundance analysis
ps_filt <- ps %>%
  filter_taxa(function(x) sum(x > 3) > 0.1*length(x), TRUE)
# Reduces to 23 taxa (wow!)

samdf <- as.data.frame(as.matrix(ps@sam_data)) %>%
  mutate(DOL = as.numeric(DOL),
         BGA = as.numeric(BGA),
         pH = as.numeric(pH),
         redox = as.numeric(redox),
         Wt = as.numeric(Wt),
         cga = as.numeric (cga),
         bw = as.numeric(bw))

```

# pH and redox
## Analyze pH and redox in context of clinical data - birth stats
```{r}
# Summarize pH and redox within infants
## Need to account for NA values in some cases
pH_redox_summary <- pH_redox %>%
  group_by(Subject, BGA, bw, matabx) %>%
  summarize(mean_pH = mean(pH, na.rm = T),
            se_pH = sd(pH, na.rm = T)/sqrt(length(which(!is.na(pH)))),
            mean_redox = mean(redox, na.rm = T),
            se_redox = sd(redox, na.rm = T)/sqrt(length(which(!is.na(redox)))))

# Linear models - pH
pH_redox %>%
  lmer(pH ~ matabx + (1|Subject), data = .) %>%
  summary()
# BGA p = 0.016
# bw p = 0.0046
# matabx p = 0.76

# Examine birth stats
pH_BGA_plot <- ggplot(pH_redox, aes(x = BGA, y = pH, color = Subject)) +
  geom_point() +
  geom_errorbar(data = pH_redox_summary, aes(x = BGA, ymin = mean_pH, ymax = mean_pH, color = Subject),
                width = 0.5, inherit.aes = F) +
  geom_errorbar(data = pH_redox_summary, aes(x = BGA, ymin = mean_pH - se_pH,
                                             ymax = mean_pH + se_pH, color = Subject),
                width = 0.25, inherit.aes = F) +
  labs(x = "Birth gestational age (weeks)", y = "Fecal pH", color = "Participant") +
  geom_textbox(data = data.frame(x=33, y = 7, label = "Linear mixed model<br>pH ~ BGA + (1 | Subject)<br>BGA *p* = 0.016"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), inherit.aes = F)

pH_bw_plot <- ggplot(pH_redox, aes(x = bw, y = pH, color = Subject)) +
  geom_point() +
  geom_errorbar(data = pH_redox_summary, aes(x = bw, ymin = mean_pH, ymax = mean_pH, color = Subject),
                width = 40, inherit.aes = F) +
  geom_errorbar(data = pH_redox_summary, aes(x = bw, ymin = mean_pH - se_pH,
                                             ymax = mean_pH + se_pH, color = Subject),
                width = 20, inherit.aes = F) +
  labs(x = "Birth weight (g)", y = "Fecal pH", color = "Participant") +
  geom_textbox(data = data.frame(x=1220, y = 7, label = "Linear mixed model<br>pH ~ bw + (1 | Subject)<br>bw *p* = 0.0046"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), inherit.aes = F)


# Linear models - redox
pH_redox %>%
  lmer(redox ~ BGA + (1|Subject), data = .) %>%
  summary()
# BGA p = 0.060
# bw p = 0.17
# matabx p = 0.87

redox_BGA_plot <- ggplot(pH_redox, aes(x = BGA, y = redox, color = Subject)) +
  geom_point() +
  geom_errorbar(data = pH_redox_summary, aes(x = BGA, ymin = mean_redox,
                                             ymax = mean_redox, color = Subject),
                width = 0.5, inherit.aes = F) +
  geom_errorbar(data = pH_redox_summary, aes(x = BGA, ymin = mean_redox - se_redox,
                                             ymax = mean_redox + se_redox, color = Subject),
                width = 0.25, inherit.aes = F) +
  labs(x = "Birth gestational age (weeks)", y = "Fecal redox potential (mV)", color = "Participant") +
  geom_textbox(data = data.frame(x=32.5, y = -400, label = "Linear mixed model<br>redox ~ BGA + (1 | Subject)<br>BGA *p* = 0.060"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), inherit.aes = F)

redox_bw_plot <- ggplot(pH_redox, aes(x = bw, y = redox, color = Subject)) +
  geom_point() +
  geom_errorbar(data = pH_redox_summary, aes(x = bw, ymin = mean_redox, ymax = mean_redox, color = Subject),
                width = 40, inherit.aes = F) +
  geom_errorbar(data = pH_redox_summary, aes(x = bw, ymin = mean_redox - se_redox,
                                             ymax = mean_redox + se_redox, color = Subject),
                width = 20, inherit.aes = F) +
  labs(x = "Birth weight (g)", y = "Fecal redox potential (mV)", color = "Participant") +
  geom_textbox(data = data.frame(x=1225, y = -420, label = "Linear mixed model<br>redox ~ bw + (1 | Subject)<br>bw *p* = 0.17"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), inherit.aes = F)

# Test BGA-bw correlation
## Doing correlation because I actually only have 11 points here, no longitudinal sampling
cor.test(pH_redox_summary$BGA, pH_redox_summary$bw, method = "spearman")
# p = 0.013, rho = 0.71

BGA_bw_plot <- ggplot(pH_redox_summary, aes(x = BGA, y = bw, color = Subject)) +
  geom_point() +
  labs(x = "Birth gestational age (weeks)", y = "Birth weight (g)", color = "Participant") +
  geom_textbox(data = data.frame(x=32.5, y = 900, label = "Spearman's &rho; = 0.71<br>*p* = 0.013"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), inherit.aes = F)

# pH and redox plot
pH_redox %>%
  lmer(redox ~ pH + (1|Subject), data = .) %>%
  summary()
# p = 0.13

pH_redox_plot <- ggplot(pH_redox, aes(x = pH, y = redox, color = Subject)) +
  geom_point() +
  labs(x = "Fecal pH", y = "Fecal redox potential (mV)", color = "Participant") +
  geom_textbox(data = data.frame(x=6.6, y = -400, label = "Linear mixed model<br>redox ~ pH + (1 | Subject)<br>pH *p* = 0.13"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), inherit.aes = F)

```

## Analyze pH and redox in context of clinical data longitudinally
```{r}
pH_time_plot <- ggplot(pH_redox, aes(x = cga, y = pH, color = Subject)) +
  geom_point() +
  geom_line()

# LMM correcting for bw
pH_redox %>%
  lmer(redox ~ bw + DOL + (1 | Subject), data  = .) %>%
  summary()
# pH
## DOL p = 0.38
## Wt p = 0.61
# redox
## DOL p = 0.0045
## Wt p = 0.27

redox_DOL_plot <- ggplot(pH_redox, aes(x = DOL, y = redox, color = Subject)) +
  geom_point() +
  geom_line(alpha = 0.4) +
  labs(x = "Age (days)", y = "Fecal redox potential (mV)", color = "Participant") +
  geom_textbox(data = data.frame(x = 60, y = -475, label = "Linear mixed model<br>redox ~ bw + DOL + (1 | Subject)<br>DOL *p* = 0.0045"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), width = unit(3, "inch"), inherit.aes = F)

pH_DOL_plot <- ggplot(pH_redox, aes(x = DOL, y = pH, color = Subject)) +
  geom_point() +
  geom_line(alpha = 0.4) +
  labs(x = "Age (days)", y = "Fecal pH", color = "Participant") +
  geom_textbox(data = data.frame(x = 64, y = 5.42, label = "Linear mixed model<br>pH ~ bw + DOL + (1 | Subject)<br>DOL *p* = 0.38"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), width = unit(3, "inch"), inherit.aes = F)


```

# SCFA
## Analyze SCFA data in context of clinical data
```{r}
# Gather to make it easier to plot all SCFAs at once
scfa_gather <- pH_redox %>%
  gather(key = scfa, value = conc, acetate:total)

# lmm
scfa_lmm_results <- data.frame(scfa = rep(unique(scfa_gather$scfa), 6),
                               var = rep(c("BGA", "bw", "pH", "redox", "DOL", "Wt"), each = 4),
                               p = NA)

for(r in 1:nrow(scfa_lmm_results)) {
  
  # For longitudinal variables (DOL, Wt), control for birth weight
  if(scfa_lmm_results$var[r] %in% c("DOL", "Wt", "feedintol")) {
    scfa_lmm_results$p[r] <- scfa_gather %>%
      filter(scfa == scfa_lmm_results$scfa[r]) %>%
      lmer(conc ~ bw + get(scfa_lmm_results$var[r]) + (1|Subject), data = .) %>%
      summary() %>%
      .$coefficients %>%
      .[3,5]
      
  } else {
    scfa_lmm_results$p[r] <- scfa_gather %>%
      filter(scfa == scfa_lmm_results$scfa[r]) %>%
      lmer(conc ~ get(scfa_lmm_results$var[r]) + (1|Subject), data = .) %>%
      summary() %>%
      .$coefficients %>%
      .[2,5]
  }

}
# all NS except:
## total ~ pH p = 0.0057, and acetate ~ pH p = 0.0030
## total ~ bw + DOL p = 0.024, and acetate ~ bw + DOL p = 0.029
## propionate ~ bw + DOL p = 0.031, and propionate ~ bw + Wt p = 0.00047

# Total/ace and pH
total_pH_plot <- ggplot(pH_redox, aes(x = pH, y = total, color = Subject)) +
  geom_point() +
  labs(x = "Fecal pH", y = "Total SCFA concentration (mM)", color = "Participant") +
  geom_textbox(data = data.frame(x=6.8, y = 120, label = "Linear mixed model<br>total ~ pH + (1 | Subject)<br>pH *p* = 0.0057"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), inherit.aes = F)

ace_pH_plot <- ggplot(pH_redox, aes(x = pH, y = acetate, color = Subject)) +
  geom_point() +
  labs(x = "Fecal pH", y = "Acetate concentration (mM)", color = "Participant") +
  geom_textbox(data = data.frame(x=6.8, y = 120, label = "Linear mixed model<br>acetate ~ pH + (1 | Subject)<br>pH *p* = 0.0030"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), inherit.aes = F)
  
# Total/ace and DOL
total_DOL_plot <- ggplot(pH_redox, aes(x = DOL, y = total, color = Subject)) +
  geom_point() +
  geom_line(alpha = 0.4) +
  labs(x = "Age (days)", y = "Total SCFA concentration (mM)", color = "Participant") +
  geom_textbox(data = data.frame(x = 60, y = 175, label = "Linear mixed model<br>total ~ bw + DOL + (1 | Subject)<br>DOL *p* = 0.024"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), width = unit(3, "inch"), inherit.aes = F)

ace_DOL_plot <- ggplot(pH_redox, aes(x = DOL, y = acetate, color = Subject)) +
  geom_point() +
  geom_line(alpha = 0.4) +
  labs(x = "Age (days)", y = "Acetate concentration (mM)", color = "Participant") +
  geom_textbox(data = data.frame(x = 60, y = 130, label = "Linear mixed model<br>acetate ~ bw + DOL + (1 | Subject)<br>DOL *p* = 0.029"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), width = unit(3, "inch"), inherit.aes = F)

# Propionate and DOL / Wt
pro_DOL_plot <- ggplot(pH_redox, aes(x = DOL, y = propionate, color = Subject)) +
  geom_point() +
  geom_line(alpha = 0.4) +
  labs(x = "Age (days)", y = "Propionate concentration (mM)", color = "Participant") +
  geom_textbox(data = data.frame(x = 58, y = 20, label = "Linear mixed model<br>propionate ~ bw + DOL + (1 | Subject)<br>DOL *p* = 0.031"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), width = unit(3, "inch"), inherit.aes = F)

pro_Wt_plot <- ggplot(pH_redox, aes(x = Wt, y = propionate, color = Subject)) +
  geom_point() +
  geom_line(alpha = 0.4) +
  labs(x = "Weight (g)", y = "Propionate concentration (mM)", color = "Participant") +
  geom_textbox(data = data.frame(x = 1200, y = 20, label = "Linear mixed model<br>propionate ~ bw + Wt + (1 | Subject)<br>Wt *p* = 0.00047"),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), width = unit(3, "inch"), inherit.aes = F)
  

```

# 16S
## Analyze 16S data in context of clinical data
```{r}
# Diversity
diversity <- ps %>%
  estimate_richness(split = TRUE, measures = c("Shannon", "Observed")) %>%
  cbind(ps@sam_data) %>%
  gather(key = measure, value = value, c("Shannon", "Observed"))

diversity_summary <- diversity %>%
  group_by(Subject_ID, bw, BGA, measure) %>%
  summarize(mean = mean(value, na.rm = T),
            se = sd(value, na.rm = T)/sqrt(length(which(!is.na(value)))))

## lmer
diversity %>%
  filter(measure == "Shannon") %>%
  lmer(value ~ bw + DOL + (1 | Subject_ID), data = .) %>%
  summary()
# Shannon bw = 0.61, BGA = 0.23
# Observed bw = 0.75, BGA = 0.50
# Shannon bw + DOL p = 9.06e-06
# Observed bw + DOL p = 0.025

## Diversity plot
diversity_bw_plot <- ggplot(diversity, aes(x = bw, y = value, color = Subject_ID)) +
  geom_point() +
  geom_errorbar(data = diversity_summary, aes(x = bw, ymin = mean, ymax = mean, color = Subject_ID),
                width = 40, inherit.aes = F) +
  geom_errorbar(data = diversity_summary, aes(x = bw, ymin = mean - se,
                                             ymax = mean + se, color = Subject_ID),
                width = 20, inherit.aes = F) +
  facet_wrap(~measure, scales = "free_y") +
  labs(x = "Birth weight (g)", y = "Alpha diversity", color = "Participant") +
  geom_textbox(data = data.frame(x = c(1100, 1220), y = c(23, 0.5),
                                 measure = c("Observed", "Shannon"),
                                 label = c("Linear mixed model<br>observed ~ bw + (1 | Subject)<br>bw *p* = 0.75", "Linear mixed model<br>Shannon ~ bw + (1 | Subject)<br>bw *p* = 0.61")),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), width = unit(3, "inch"), inherit.aes = F)
  

diversity_DOL_plot <- ggplot(diversity, aes(x = DOL, y = value,
                                            color = Subject_ID, group = Subject_ID)) +
  geom_point() +
  geom_line(alpha = 0.4) +
  facet_wrap(~measure, scales = "free_y") +
  labs(x = "Age (days)", y = "Alpha diversity", color = "Participant") +
  geom_textbox(data = data.frame(x = c(50, 60), y = c(23, 0.11),
                                 measure = c("Observed", "Shannon"),
                                 label = c("Linear mixed model<br>observed ~ bw + DOL + (1 | Subject)<br>DOL *p* = 0.025", "Linear mixed model<br>Shannon ~ bw + DOL + (1 | Subject)<br>DOL *p* = 9.1 x 10<sup>-6</sup>")),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), width = unit(3, "inch"), inherit.aes = F)

# NMDS
bray <- distance(ps, method = "bray") %>% as.matrix()

## Test effect of participant
adonis2(formula = bray ~ Subject_ID, data = samdf,
        permutations=9999, method = "bray")
# participant p < 0.0001

# More complicated model
adonis2(formula = bray ~ bw + DOL + pH + redox + Subject_ID, data = samdf,
        permutations=9999, method = "bray", na.action = na.omit)
#            Df SumOfSqs      R2       F Pr(>F)    
# bw          1   2.6453 0.10374 17.6512 0.0001 ***
# DOL         1   2.0974 0.08225 13.9951 0.0001 ***
# pH          1   0.4445 0.01743  2.9659 0.0085 ** 
# redox       1   0.4637 0.01818  3.0942 0.0054 ** 
# Subject_ID  9  10.5577 0.41403  7.8276 0.0001 ***

ord <- ordinate(ps, method="NMDS", distance="bray")

ord_plot <- plot_ordination(ps, ord, color="Subject_ID", title=NULL) +
  stat_ellipse(linetype = "dashed") +
  labs(color = "Participant") +
   geom_textbox(data = data.frame(x = 3, y = 1.25,
                                  label = paste("PERMANOVA",
                                                "bw R<sup>2</sup> = 0.10, *p* < 0.0001",
                                                "DOL R<sup>2</sup> = 0.082, *p* < 0.0001",
                                                "pH R<sup>2</sup> = 0.017, *p* = 0.0085",
                                                "redox R<sup>2</sup> = 0.018, *p* = 0.0054",
                                                "participant R<sup>2</sup> = 0.41, *p* < 0.0001",
                                                sep = "<br>")),
               aes(x, y, label = label), halign = 0.5, fill = NA, box.colour = NA,
               box.padding = grid::unit(rep(0, 4), "pt"), width = unit(3, "inch"), inherit.aes = F)


```

## Look at specific taxa with ALDEx2
```{r}
# ALDEx2 GLM incorporating multiple variables
## Using ps_filt here!
otu_flip <- as.data.frame(t(ps_filt@otu_table))
mm <- samdf[rownames(as.data.frame(as.matrix(ps_filt@sam_data))),] %>%
  model.matrix(~ bw + DOL + pH + redox + Subject_ID, data = .)

# ALDEx2 correlation (not incorporating participant...)
clr <- aldex.clr(otu_flip[,colnames(otu_flip) %in% rownames(mm)], mm, mc.samples = 128, denom ="all")
glm.test <- aldex.glm(clr) # slow
#write.csv(glm.test, "plots/aldex2_glm_bw_DOL_pH_redox_participant.csv")
# bw seq_1 p.BH = 0.0067, seq_4 = 0.033
# DOL seq_1 p.BH = 0.00028
# none p < 0.05 for pH or redox

melty <- ps_filt %>%
  microbiome::transform(transform = "compositional") %>%
  psmelt() %>%
  mutate(OTU = sub("seq_", "ASV ", OTU)) %>%
  mutate(binom = case_when(is.na(Genus) ~ paste0(Family, " family (", OTU, ")"),
                           !is.na(Genus) & is.na(Species) ~
                             paste0("*", Genus, " sp.* (", OTU, ")"),
                           !is.na(Species) ~
                             paste0("*", Genus, " ", Species, "* (", OTU, ")"))) %>%
  filter(OTU %in% c("ASV 1", "ASV 4")) %>%
  mutate(binom = factor(binom, levels = c("*Staphylococcus sp.* (ASV 1)",
                                           "*Escherichia-Shigella sp.* (ASV 4)")))

melty_summary <- melty %>%
  group_by(OTU, binom, Subject_ID, bw) %>%
  summarize(mean = mean(Abundance), se = sd(Abundance)/sqrt(length(Abundance)))

tax_bw_plot <- melty %>%
  filter(OTU %in% c("ASV 1", "ASV 4")) %>%
  ggplot(aes(x = bw, y = Abundance, color = Subject_ID)) +
  geom_point() +
  geom_errorbar(data = filter(melty_summary, OTU %in% c("ASV 1", "ASV 4")),
                aes(x = bw, ymin = mean, ymax = mean, color = Subject_ID),
                width = 40, inherit.aes = F) +
  geom_errorbar(data = filter(melty_summary, OTU %in% c("ASV 1", "ASV 4")),
                aes(x = bw, ymin = mean - se, ymax = mean + se, color = Subject_ID),
                width = 20, inherit.aes = F) +
  labs(x = "Birth weight (g)", y = "Relative abundance", color = "Participant") +
  facet_wrap(~binom, scales = "free_y", nrow = 2) +
  theme(strip.text = element_markdown())

tax_DOL_plot <- melty %>%
  filter(OTU %in% c("ASV 1")) %>%
  ggplot(aes(x = DOL, y = Abundance, color = Subject_ID)) +
  geom_point() +
  geom_line() +
  labs(x = "Age (days)", y = "Relative abundance") +
  facet_wrap(~binom, scales = "free_y") +
  theme(strip.text = element_markdown())


# Also do for genus/family
ps_genus_filt <- ps %>%
  tax_glom(taxrank = "Genus") %>%
  filter_taxa(function(x) sum(x > 3) > 0.1*length(x), TRUE)
# 16 genera

otu_flip_gen <- as.data.frame(t(ps_genus_filt@otu_table))
mm_gen <- samdf[rownames(as.data.frame(as.matrix(ps_genus_filt@sam_data))),] %>%
  model.matrix(~ bw + DOL + pH + redox + Subject_ID, data = .)

# ALDEx2 correlation (not incorporating participant...)
clr_gen <- aldex.clr(otu_flip_gen[,colnames(otu_flip_gen) %in% rownames(mm_gen)],
                     mm_gen, mc.samples = 128, denom ="all")
glm.test_gen <- aldex.glm(clr_gen) # slow
# bw ASV 1 Staphylococcus (p = 0.0015), ASV 8 Clostridium sensu stricto 1 (p = 0.019),
## ASV 4 Escherichia-Shigella (p = 0.023)
# DOL ASV 1 Staphylococcus (p = 6.0e-5)
# none for pH/redox

## Family
ps_fam_filt <- ps %>%
  tax_glom(taxrank = "Family") %>%
  filter_taxa(function(x) sum(x > 3) > 0.1*length(x), TRUE)
# 12 families

otu_flip_fam <- as.data.frame(t(ps_fam_filt@otu_table))
mm_fam <- samdf[rownames(as.data.frame(as.matrix(ps_fam_filt@sam_data))),] %>%
  model.matrix(~ bw + DOL + pH + redox + Subject_ID, data = .)

# ALDEx2 correlation (not incorporating participant...)
clr_fam <- aldex.clr(otu_flip_fam[,colnames(otu_flip_fam) %in% rownames(mm_fam)],
                     mm_fam, mc.samples = 128, denom ="all")
glm.test_fam <- aldex.glm(clr_fam) # slow
# bw ASV 1 Staphylococcaceae (p = 0.00091), ASV 8 Clostridiaceae (p = 0.0093)
# DOL ASV 1 Staphylococcaceae (p = 0.000034), ASV 3 Enterobacteriaceae (p = 0.0011)

melty_fam <- ps_fam_filt %>%
  microbiome::transform(transform = "clr") %>%
  psmelt()

fam_bw_plot <- melty_fam %>%
  filter(OTU %in% c("seq_1", "seq_8")) %>%
  ggplot(aes(x = bw, y = Abundance, color = Subject_ID)) +
  geom_point() +
  labs(x = "Birth weight (g)", y = "Relative abundance") +
  facet_wrap(~Family, scales = "free_y")

fam_DOL_plot <- melty_fam %>%
  filter(OTU %in% c("seq_1", "seq_3")) %>%
  ggplot(aes(x = DOL, y = Abundance, color = Subject_ID)) +
  geom_point() +
  geom_line() +
  labs(x = "Age (days)", y = "Relative abundance") +
  facet_wrap(~Family, scales = "free_y")

```


# Can we predict pH / redox from 16S with random forest?
# what about feedintol?
```{r}
# lm
ps_filt_clr <- ps_filt %>% # additional feature selection
  microbiome::transform(transform = "clr")

clr_spread <- cbind(ps_filt_clr@sam_data, ps_filt_clr@otu_table)

# paste(colnames(melty_spread)[11:33], collapse = " + ")

clr_spread %>%
  lmer(redox ~ seq_1 + seq_2 + seq_3 + seq_4 + seq_5 + seq_6 + seq_7 + seq_8 + seq_9 + seq_10 +
         seq_11 + seq_12 + seq_13 + seq_15 + seq_16 + seq_17 + seq_19 + seq_23 + seq_24 +
         seq_25 + seq_26 + seq_30 + seq_33 + bw + DOL + (1 | Subject_ID), data = .) %>%
  summary()
# pH: none p < 0.05
# redox: seq_2 p = 0.0368

# seq_15



# by family
ps_fam_filt_clr <- ps_fam_filt %>%
  microbiome::transform(transform = "clr")

fam_clr_spread <- cbind(ps_fam_filt_clr@sam_data, ps_fam_filt_clr@otu_table)

#paste(colnames(fam_clr_spread)[11:22], collapse = " + ")

fam_clr_spread %>%
  lmer(redox ~ seq_1 + seq_2 + seq_3 + seq_6 + seq_7 + seq_8 + seq_12 + seq_13 +
         seq_15 + seq_17 + seq_20 + seq_26 + bw + DOL + (1 | Subject_ID), data = .) %>%
  summary()


```

# Patch figures
```{r}
# Fig. 1 - birth stats and pH/redox
fig1 <- plot_grid(pH_bw_plot + theme(legend.position = "none"),
                  redox_bw_plot + theme(legend.position = "none"),
                  pH_BGA_plot + theme(legend.position = "none"),
                  redox_BGA_plot + theme(legend.position = "none"),
                  pH_DOL_plot + theme(legend.position = "none"),
                  redox_DOL_plot + theme(legend.position = "none"),
                  ncol = 2,
                  labels = c("A", "B", "C", "D", "E", "F"), label_fontface = "plain") %>%
  plot_grid(., get_legend(pH_BGA_plot), rel_widths = c(8,1))
#ggsave("plots/fig1.png", fig1, height = 10, width = 10)

# Fig. 2 - SCFA
fig2 <- plot_grid(total_pH_plot + theme(legend.position = "none"),
                  ace_pH_plot + theme(legend.position = "none"),
                  total_DOL_plot + theme(legend.position = "none"),
                  ace_DOL_plot + theme(legend.position = "none"),
                  pro_DOL_plot + theme(legend.position = "none"),
                  pro_Wt_plot + theme(legend.position = "none"),
                  ncol = 2,
                  labels = c("A", "B", "C", "D", "E", "F"), label_fontface = "plain") %>%
  plot_grid(., get_legend(total_pH_plot), rel_widths = c(8,1))
#ggsave("plots/fig2.png", fig2, height = 10, width = 10)

# Fig. 3 16S
fig3 <- plot_grid(ord_plot + theme(legend.position = "none"),
                       tax_DOL_plot + theme(legend.position = "none"), 
                       ncol = 1, labels = c("A", "C"), label_fontface = "plain") %>%
  plot_grid(., tax_bw_plot,
            labels = c("", "B"), label_fontface = "plain")
#ggsave("plots/fig3.png", fig3, height = 8, width = 10)

# Fig. 4 random forest?


# Fig. S1 
figS1 <- plot_grid(BGA_bw_plot + theme(legend.position = "none"),
                   pH_redox_plot  + theme(legend.position = "none"),
                   get_legend(pH_redox_plot),
                   nrow = 1, rel_widths = c(5,5,1),
                   labels = c("A", "B", ""), label_fontface = "plain")
#ggsave("plots/figS1.png", figS1, height = 4, width = 8)

# Fig. S2
figS2 <- plot_grid(diversity_bw_plot + theme(legend.position = "none"),
                   diversity_DOL_plot + theme(legend.position = "none"),
                   nrow = 2, labels = c("A", "B"), label_fontface = "plain") %>%
  plot_grid(., get_legend(diversity_DOL_plot),
            nrow = 1, rel_widths = c(10, 1),
            labels = c("", ""), label_fontface = "plain")
#ggsave("plots/figS2.png", figS2, height = 8, width = 10)





```